sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -624,17 +611,40 @@ def _load_configuration_settings(self, **kwargs):
                     sentinel_keys[(config.key, config.label)] = config.etag
         return configuration_settings, sentinel_keys
 
+    @staticmethod
+    def _calculate_feature_id(key, label):
+        basic_value = f"{key}\n"
+        if label:
+            basic_value = basic_value + f"\n{label}"
+        feature_flag_id_hash_bytes = hashlib.sha256(basic_value.encode()).digest()
+        encoded_flag = base64.b64encode(feature_flag_id_hash_bytes)
+        encoded_flag = encoded_flag.replace(b"+", b"-").replace(b"/", b"_")
+        return encoded_flag[: encoded_flag.find(b"=")]
+
     def _load_feature_flags(self, **kwargs):
         feature_flag_sentinel_keys = {}
         loaded_feature_flags = []
         # Needs to be removed unknown keyword argument for list_configuration_settings
         kwargs.pop("sentinel_keys", None)
         for select in self._feature_flag_selectors:
+            endpoint = self._client._impl._config.endpoint  # pylint: disable=protected-access
             feature_flags = self._client.list_configuration_settings(
                 key_filter=FEATURE_FLAG_PREFIX + select.key_filter, label_filter=select.label_filter, **kwargs
             )
             for feature_flag in feature_flags:
-                loaded_feature_flags.append(json.loads(feature_flag.value))
+                feature_flag_value = json.loads(feature_flag.value)
+                if "telemetry" in feature_flag_value:
+                    if "metadata" not in feature_flag_value["telemetry"]:
+                        feature_flag_value["telemetry"]["metadata"] = {}
+                    feature_flag_value["telemetry"]["metadata"]["etag"] = feature_flag.etag
+                    feature_flag_reference = f"{endpoint}/kv/{feature_flag.key}"
```

Comment:
can endpoint be null?

Reply:
No, not here. If you look above, I'm getting the endpoint from the SDK client. Side note, this is only temporary. I'm working on the discovery/auto failover PR where this will be managed by us.